bpf/lib/common.h

```diff
@@ -297,7 +297,7 @@ struct policy_entry {
 	__be16		proxy_port;
 	__u8		deny:1,
 			pad:7;
-	__u8		pad0;
+	__u8		auth_type;
```

Comment:
IMO we need to pass the requested auth type to the datapath (e.g., SPIFFE) as we need to know which auth agent should perform the auth upon the auth requested notification. We could also do a search on the ep's policy to find this info, but that would be more overhead.

Reply:
I think this sorta comes down to whether we actually intend on supporting multiple auth types at the same time. Do we have use cases to redirect the individual connections for some pods to different auth agents at the same time? Setting up the whole authentication infrastructure is complicated, I'd be surprised to find that there's a solid argument for why someone would be interested in doing that multiple times with different auth protocols. I'm not sure we need to impose this requirement on ourselves. We make the CNP level configurable so that we have the _option_ to extend the structure later on, but down here at the datapath layer we likely have more flexibility around changing the details later.

Reply:
I'm not sure, but that would facilitate auth agents used for testing only at the same time with a real thing. Also passing a byte down and back up without the datapath trying to interpret its meaning would enable more than a simple enum without any datapath changes (e.g. Use 4 bits as an enum and 4 bits as related flags)? If we end up only using a few bits out of the byte we can later change to use the unused bits in the byte for something else.

On a related note I feel we need more than one byte for the conntrack, e.g., auth needed/auth granted. With only one bit we have a problem where an existing connections look like "auth granted (auth_required bit is cleared) when policy flips from no auth to auth required. May also need an auth expiry timestamp, but that would require extending the CT map value that I'd like not to address at this stage.

Reply:
I agree it would facilitate multiple types at the same time, I just don't think we need to support that. This is the sort of stuff that I'm sure we'll adapt over time but at this point I don't see the need for it. Maybe once we have the full control plane setup on top it'll become more clear.

I still don't follow the need for conntrack involvement at all in the longer term direction, but yeah I don't think we need to address this at this stage. Let's make sure to follow up on that design discussion in the new year and understand the interactions. Maybe you already have a much better idea of the interaction there than I do, I just need to catch up on it :)